qualcommax: ipq60xx: add TP-Link EAP625-Outdoor support.

jmspswny opened the pull request:
TP-Link EAP625-Outdoor PoE only, wall / pole mountable, external antenna AP that is marketed as AX1800.

Specifications:
---------------
* CPU: Qualcomm IPQ6018 Quad core Cortex-A53 (64-bit Quad-core Arm Cortex-A53 @ 1800MHz)
* RAM: 1024 MB
* Storage: GigaDevice PSR1GA30DT 128MB NAND
* Ethernet: Realtek RTL8211F Gigabit RJ45 port with PoE input
* Wi-Fi: QCN9074 (4x4 5 GHz 802.11ax)
* Wi-Fi: IPQ6018 (4x4 2.4 GHz 802.11b/g/n/ax)
* Wi-Fi: 2 external RP-SMA antennas
* LEDs: 1x Green Status (GPIO 37 Active High), 1x Yellow Status (GPIO 32 Active High) and an LED global control GPIO (GPIO 36 Active High, set up by U-Boot)
* Buttons: 1x Reset (GPIO 9 Active Low)
* Serial Port / UART: 4-pin unpopulated header. Pinout 1 - TX, 2 - RX, 3 - GND, 4 - VCC 115200baud

Installation (instructions cribbed from the EAP610-Outdoor, courtesy mrnuke):

Web UI method
-------------

Set up the device using the vendor's web UI. After that go to Management->SSH and enable the "SSH Login" checkbox. Select "Save". The connect to the machine via SSH:

ssh -o hostkeyalgorithms=ssh-rsa <ip_of_device>

Disable signature verification:

cliclientd stopcs

Rename the "-web-ui-factory" image to something less than 63 characters, maintaining the ".bin" suffix.
* Go to System -> Firmware Update.
* Under "New Firmware File", click "Browse" and select the image
* Select "Update" and confirm by clicking "OK".

If the update fails, the web UI should show an error message. Otherwise, the device should reboot into OpenWRT.

TFTP method
-----------

To flash via tftp, first place the initramfs image on the TFTP server, then access the device via serial connection. Once connected, stop the boot process point u-boot at your tftp server:

setenv serverip <ip of tftp server>
setenv ipaddr <ip in same subnet as tftp server>
tftpboot tplink_eap610-outdoor-initramfs-uImage.itb bootm

This should boot OpenWRT. Once booted, flash the sysupgrade.bin image using either luci or the commandline.

Signed-off-by: James Sweeney <code@swny.io>

jmspswny commented:
the board  firmware PR is here: https://github.com/openwrt/firmware_qca-wireless/pull/82

robimarko commented:
Please fixup the formal issues, also please fix the commit title as it currently includes part of the description as well

185264646 commented:
I don't think you should remove `HD v1` suffix.  From TP-Link official website:
> EAP623-Outdoor HD is based on EAP610-Outdoor and adds a set of auxiliary antennas to combat outdoor interference.

clearly HD is part of the name. Please don't remove it. It's different from EAP610.

robimarko commented on package/boot/uboot-tools/uboot-envtools/files/qualcommax_ipq60xx:
```diff
@@ -29,7 +29,8 @@ netgear,wax214|\
 netgear,wax610|\
 netgear,wax610y|\
 tplink,eap610-outdoor|\
-tplink,eap623od-hd-v1)
```
As @185264646 already pointed out, DO NOT remove `tplink,eap623od-hd-v1`

robimarko commented:
Why are you renaming the board?

In the current commit, this will break sysupgrading for all current users.

jmspswny commented:
because the current board name creates an arbitrary, non-standard abbreviation by utilizing "od" for "outdoor".  Changing the name now, 8 days after the board was added, should be minimally impactful.  If it waits much longer, then it should just remain as is.

robimarko commented:
> because the current board name creates an arbitrary, non-standard abbreviation by utilizing "od" for "outdoor". Changing the name now, 8 days after the board was added, should be minimally impactful. If it waits much longer, then it should just remain as is, leaving subsequent users to wonder why a device named "EAP623-Outdoor" on the packaging is now "EAP623OD"

In that case please split up the changes, rename the existing board (But make sure to utilize SUPPORTED_DEVICES to provide support for the existing name) in one commit, and then generalize the stuff in the second one and only in the third one add the new device.

Commit:
qualcommax: ipq60xx: add TP-Link EAP625-Outdoor support.

TP-Link EAP625-Outdoor PoE only, wall / pole mountable, external antenna AP that is marketed as AX1800.

Specifications:
---------------
* CPU: Qualcomm IPQ6018 Quad core Cortex-A53 (64-bit Quad-core Arm Cortex-A53 @ 1800MHz)
* RAM: 1024 MB
* Storage: GigaDevice PSR1GA30DT 128MB NAND
* Ethernet: Realtek RTL8211F Gigabit RJ45 port with PoE input
* Wi-Fi: QCN9074 (4x4 5 GHz 802.11ax)
* Wi-Fi: IPQ6018 (4x4 2.4 GHz 802.11b/g/n/ax)
* Wi-Fi: 2 external RP-SMA antennas
* LEDs: 1x Green Status (GPIO 37 Active High), 1x Yellow Status (GPIO 32 Active High) and an LED global control GPIO (GPIO 36 Active High, set up by U-Boot)
* Buttons: 1x Reset (GPIO 9 Active Low)
* Serial Port / UART: 4-pin unpopulated header. Pinout 1 - TX, 2 - RX, 3 - GND, 4 - VCC 115200baud

Installation (instructions cribbed from the EAP610-Outdoor, courtesy mrnuke):
=============

Web UI method
-------------

Set up the device using the vendor's web UI. After that go to
Management->SSH and enable the "SSH Login" checkbox. Select "Save".
The connect to the machine via SSH:

ssh -o hostkeyalgorithms=ssh-rsa <ip_of_device>

Disable signature verification:

cliclientd stopcs

Rename the "-web-ui-factory" image to something less than 63
characters, maintaining the ".bin" suffix.
* Go to System -> Firmware Update.
* Under "New Firmware File", click "Browse" and select the image
* Select "Update" and confirm by clicking "OK".

If the update fails, the web UI should show an error message.
Otherwise, the device should reboot into OpenWRT.

TFTP method
-----------

To flash via tftp, first place the initramfs image on the TFTP
server, then access the device via serial connection. Once
connected, stop the boot process point u-boot at your tftp server:

setenv serverip <ip of tftp server>
setenv ipaddr <ip in same subnet as tftp server>
tftpboot tplink_eap610-outdoor-initramfs-uImage.itb
bootm

This should boot OpenWRT. Once booted, flash the sysupgrade.bin
image using either luci or the commandline.

Signed-off-by: James Sweeney <code@swny.io>

185264646 commented:
> because the current board name creates an arbitrary, non-standard abbreviation by utilizing "od" for "outdoor". Changing the name now, 8 days after the board was added, should be minimally impactful. If it waits much longer, then it should just remain as is, leaving subsequent users to wonder why a device named "EAP623-Outdoor" on the packaging is now "EAP623OD"

To make things clear, please download the GPL code from https://static.tp-link.com/upload/gpl-code/2024/202404/20240424/EAP625-OutdoorHDv1.6-gpl.tar.gz . Then uncompress the code and navigate to `eap_gpl_code/build/product_configs/ipq60xx_eap623od_625od_common`. So `od` for `outdoor` is not an arbitrary abbriviation, but what they call this product internally. Besides this, it's clear this poduct is called `EAP623-Outdoor HD` and it's v1. I can understand why you rename od->outdoor, but you still haven't tell us the reason to remove the `hd-v1` suffix.

jmspswny commented:
> > because the current board name creates an arbitrary, non-standard abbreviation by utilizing "od" for "outdoor". Changing the name now, 8 days after the board was added, should be minimally impactful. If it waits much longer, then it should just remain as is, leaving subsequent users to wonder why a device named "EAP623-Outdoor" on the packaging is now "EAP623OD"
> 
> To make things clear, please download the GPL code from https://static.tp-link.com/upload/gpl-code/2024/202404/20240424/EAP625-OutdoorHDv1.6-gpl.tar.gz . Then uncompress the code and navigate to `eap_gpl_code/build/product_configs/ipq60xx_eap623od_625od_common`. So `od` for `outdoor` is not an arbitrary abbriviation, but what they call this product internally. Besides this, it's clear this poduct is called `EAP623-Outdoor HD` and it's v1. I can understand why you rename od->outdoor, but you still haven't tell us the reason to remove the `hd-v1` suffix.

your commentary is no longer relevant to this PR.

185264646 commented:
> > > because the current board name creates an arbitrary, non-standard abbreviation by utilizing "od" for "outdoor". Changing the name now, 8 days after the board was added, should be minimally impactful. If it waits much longer, then it should just remain as is, leaving subsequent users to wonder why a device named "EAP623-Outdoor" on the packaging is now "EAP623OD"
> > 
> > 
> > To make things clear, please download the GPL code from https://static.tp-link.com/upload/gpl-code/2024/202404/20240424/EAP625-OutdoorHDv1.6-gpl.tar.gz . Then uncompress the code and navigate to `eap_gpl_code/build/product_configs/ipq60xx_eap623od_625od_common`. So `od` for `outdoor` is not an arbitrary abbriviation, but what they call this product internally. Besides this, it's clear this poduct is called `EAP623-Outdoor HD` and it's v1. I can understand why you rename od->outdoor, but you still haven't tell us the reason to remove the `hd-v1` suffix.
> 
> your commentary is no longer relevant to this PR.

The same thing also applies to `eap625`. The official name contains **HD**, but you simply removed it without clear reason. Anyway, it's your code, so i'm fine with it.

By the way, the common dtsi should only applies to 623 and 625, so the name `62x-common` is not appropriate. afaik, `eap620hd v1` uses ipq8070a and should not use this dtsi.

jmspswny commented:
> > > > because the current board name creates an arbitrary, non-standard abbreviation by utilizing "od" for "outdoor". Changing the name now, 8 days after the board was added, should be minimally impactful. If it waits much longer, then it should just remain as is, leaving subsequent users to wonder why a device named "EAP623-Outdoor" on the packaging is now "EAP623OD"
> > > 
> > > 
> > > To make things clear, please download the GPL code from https://static.tp-link.com/upload/gpl-code/2024/202404/20240424/EAP625-OutdoorHDv1.6-gpl.tar.gz . Then uncompress the code and navigate to `eap_gpl_code/build/product_configs/ipq60xx_eap623od_625od_common`. So `od` for `outdoor` is not an arbitrary abbriviation, but what they call this product internally. Besides this, it's clear this poduct is called `EAP623-Outdoor HD` and it's v1. I can understand why you rename od->outdoor, but you still haven't tell us the reason to remove the `hd-v1` suffix.
> > 
> > 
> > your commentary is no longer relevant to this PR.
> 
> The same thing also applies to `eap625`. The official name contains **HD**, but you simply removed it without clear reason. Anyway, it's your code, so i'm fine with it.
> 
> By the way, the common dtsi should only applies to 623 and 625, so the name `62x-common` is not appropriate. afaik, `eap620hd v1` uses ipq8070a and should not use this dtsi.

Well, if that was the dtsi file name, then your point might be valid.  As it is `6XX-outdoor-common`, I'm not sure why someone would think it was for one of the non-outdoor devices.

185264646 commented:
> > > > > because the current board name creates an arbitrary, non-standard abbreviation by utilizing "od" for "outdoor". Changing the name now, 8 days after the board was added, should be minimally impactful. If it waits much longer, then it should just remain as is, leaving subsequent users to wonder why a device named "EAP623-Outdoor" on the packaging is now "EAP623OD"
> > > > 
> > > > 
> > > > To make things clear, please download the GPL code from https://static.tp-link.com/upload/gpl-code/2024/202404/20240424/EAP625-OutdoorHDv1.6-gpl.tar.gz . Then uncompress the code and navigate to `eap_gpl_code/build/product_configs/ipq60xx_eap623od_625od_common`. So `od` for `outdoor` is not an arbitrary abbriviation, but what they call this product internally. Besides this, it's clear this poduct is called `EAP623-Outdoor HD` and it's v1. I can understand why you rename od->outdoor, but you still haven't tell us the reason to remove the `hd-v1` suffix.
> > > 
> > > 
> > > your commentary is no longer relevant to this PR.
> > 
> > 
> > The same thing also applies to `eap625`. The official name contains **HD**, but you simply removed it without clear reason. Anyway, it's your code, so i'm fine with it.
> > By the way, the common dtsi should only applies to 623 and 625, so the name `62x-common` is not appropriate. afaik, `eap620hd v1` uses ipq8070a and should not use this dtsi.
> 
> Well, if that was the dtsi file name, then your point might be valid. As it is `6XX-outdoor-common`, I'm not sure why someone would think it was for one of the non-outdoor devices.

`62x-outdoor-common.dtsi` might makes sense. But that's because 623 and 625 are currently the only two 62x products released so far. They (TP-Link) makes no guarantee that eap62x is only for ipq6018 and follows the same gpio and led configuration. `6xx` is surely the wrong name. since `eap650 outdoor v1`(no HD) uses ipq5018.

185264646 commented on target/linux/qualcommax/image/ipq60xx.mk:
```diff
@@ -183,6 +206,13 @@ define Device/tplink_eap623od-hd-v1
 
+define Device/tplink_eap625-outdoor
+	$(Device/tplink_eap6XX-outdoor-base)
+	DEVICE_MODEL := EAP625-Outdoor
```
Please at least add `HD` suffix to the full name here. You can use whatever short name you like as long as it's not ambiguous. But please make full name accurate and matches official name.

185264646 commented on package/firmware/ipq-wifi/Makefile:
```diff
@@ -218,6 +219,7 @@ $(eval $(call generate-ipq-wifi-package,spectrum_sax1v1k,Spectrum SAX1V1K))
 $(eval $(call generate-ipq-wifi-package,tplink_eap610-outdoor,TPLink EAP610-Outdoor))
 $(eval $(call generate-ipq-wifi-package,tplink_eap620hd-v1,TP-Link EAP620 HD v1))
 $(eval $(call generate-ipq-wifi-package,tplink_eap623od-hd-v1,TP-Link EAP623-Outdoor HD v1))
+$(eval $(call generate-ipq-wifi-package,tplink_eap625-outdoor,TP-Link EAP625-Outdoor))
```
Same here. `v1` can be optional. but better still have `HD`.

I still suggest adding v1 here. So we won't get into trouble when they release v2.

jmspswny commented:
Based on the way the TP-Link employee @185264646 has chosen to engage on this PR, I'm just closing it. I'll be spending my time and my money on other hardware.

185264646 commented:
@jmspswny I can understand why you temped and now apologize to you. But:

1. I haven't said i'm an TP-Link employee. I'm just an OpenWRT contributor and enthusiast here.
2. This is just a technical debate. We all want to make the community and the project better.
3. The naming schema can be indicated from many places like TP-link official website(URLs) and the GPL code released by them. I don't really care about `od` or `outdoor` stuff. But i think removing `HD` is not approapriate.
4. The same logic applies to `dtsi` naming. I simply give my suggestion here. And i'm not forcing you (nor do i have the right) to apply my suggestion.
5. When and whether this pr can be merged is not up to me, but the maintainers. They have the right to make the final decision.
6. It's your own decision whether to continue contributing to OpenWRT or to buy TP-Link products. I respect your decision.

jmspswny commented:
@185264646 that's a lot of buts.

1. your use of a corporate TP-Link email address for your contributions suggests otherwise.  
2. the points you've brought up are marketing, at best. to boot, your prior commits and commentary prioritize internal abbreviations over human readable information.  `od` is not a generally accepted abbreviation for `outdoor`.  `HD` in this context apparently means `high density` and doesn't provide any differentiation beyond the EAPxxx name, it just repurposes an abbreviation which the public associates with "better".

So, have at it.  Unreadable device names justified with GPL release archives code will continue. You can, in whatever relationship with TP-Link you care to evince, make sure marketingese is constantly applied.   

I'll continue contributing to OpenWRT, as the ability to choose hardware independent of the software is one of the project's fundamental strengths.

robimarko commented:
@jmspswny I think you are overreacting in this case, anybody can (And should) comment on points they are interested in

185264646 commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-eap6XX-outdoor-base.dtsi:
```diff
@@ -0,0 +1,142 @@
+		led_sys_yellow: led-1 {
+			color = <LED_COLOR_ID_YELLOW>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&tlmm 32 GPIO_ACTIVE_HIGH>;
```
Please re-check this part. The GPL code suggests that EPA625 and EAP623 are not sharing the same pins for LEDs
